src/frontends/onnx/frontend/src/op/org.openvinotoolkit/generate_proposals.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "generate_proposals.hpp"
+
+#include "default_opset.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+
+namespace {
+void validate_generate_proposals_inputs(const OutputVector& inputs) {
+    OPENVINO_ASSERT(inputs.size() == 4, "GenerateProposals operator expects 4 inputs, got ", inputs.size());
+
+    const auto scores_rank = inputs[0].get_partial_shape().rank();
+    OPENVINO_ASSERT(scores_rank.compatible(4), "GenerateProposals input scores rank should be 4, is ", scores_rank);
```

Comment:
Can you check if this is verified in the operator's shape inference function? Perhaps then it could be removed from here.

Reply:
Yes, it is. However I put here intentionally, as these assumptions are crucial for subgraph definition by ONNX FE.
In fact it's not needed at current state. I'll comment these with appropriate comment.